src/js/experiments/snippetFeedback.js

```diff
@@ -0,0 +1,261 @@
+          textForm.value = textForm.value.substring(
+            0,
+            SnippetFeedback.MAX_CHAR_COUNT - 1,
+          );
+        }
+
+        // every time the user types we need to re-render the dom content manually.
+        this._renderCharCount();
+      });
+
+      const sendButton = this._makeElement({
+        kind: 'button',
+        className: 'text-btn',
+        text: 'Send',
+        onClick: () => {
+          this._trackFormSubmission(textForm.value);
+          this.wasThisHelpfulContainer.innerHTML =
+            '<div>Thank you for your feedback!</div>';
+        },
+      });
+
+      // build the dom nodes.
+      container.appendChild(prompt);
+      container.appendChild(textForm);
+      bottomRow.appendChild(this.currCharCountElement);
+      bottomRow.appendChild(sendButton);
+      container.appendChild(bottomRow);
+      this.feedbackForm = container;
+      this.wasThisHelpfulContainer.appendChild(container);
+
+      // Here we setup the document event listeners that handle hiding the form
+      // when the user clicks outside of it.
+      //
+      // Why do we use a setTimeout with a timer of 0?
+      // -> Because there is a race condition with the adding of the click
+      // listeners to the document that immediately close the form when it opens
+      //
+      // Setting it to 0 causes the inner callback to be run asynchronously, adding it to the eventqueue
+      // so that it executes after the form being opened.
+      setTimeout(() => {
+        const outsideClickListener = (event) => {
+          if (!this.feedbackForm.contains(event.target)) {
+            this.wasThisHelpfulContainer.removeChild(this.feedbackForm);
+            this.feedbackForm = null;
+            removeClickListener();
+          }
+        };
+
+        const removeClickListener = () => {
+          document.removeEventListener('click', outsideClickListener);
+        };
+
+        document.addEventListener('click', outsideClickListener);
+      }, 0);
+    }
+  }
+
+  /**
+   * Render the char count and set it's result on the respective element.
+   * */
+  _renderCharCount() {
+    let charCount = this.currCharCount;
+    let charCountLimited =
+      charCount > SnippetFeedback.MAX_CHAR_COUNT
+        ? SnippetFeedback.MAX_CHAR_COUNT
+        : charCount;
+    const el = `
+        <div class="charCountRow">
+          <span>${charCountLimited} / ${SnippetFeedback.MAX_CHAR_COUNT}</span>
+        </div>`;
+    this.currCharCountElement.innerHTML = el;
+  }
+
+  // -- Trackers --
+  // These are responsible for sending data based on user input
+  // to amplitude.
+
+  /**
+   * _trackYesOrNoButton sends a trackAction to aplitude.
+   * actions are: "docs-snippet-helpful-no"  and "docs-snippet-helpful-yes"
+   * */
+  _trackYesOrNoButton(yesOrNoString) {
+    const date = new Date();
+    window.AnalyticsClient.trackAction(
+      `docs-snippet-helpful-${yesOrNoString}`,
```

Comment:
I think we should keep everything into one event and have the yes/no as a meta:

`docs-snippet-feedback, {helpful: true}`
`docs-snippet-feedback, {helpful: false}`

It would be easier to track how much feedback we got and not hard to separate the data between helpful/not helpful either. Thoughts?

Reply:
I'm not sure. I imagine that many users will decline to send written feedback. If we track yes/no, at least we know how many clicked it ? If there is an associated cost that we're paying per event tracked I understand why we might not want to do it, but if not why not track it and then just not use it if it's not useful?

Reply:
@teesloane What I am suggesting is just for the Click on `Yes` | `No`. Basically, group them under the same event name.

Reply:
Oh! Sorry, I misunderstood. I'll fix that.